snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/controller/TrackerController.kt

```diff
@@ -116,4 +120,23 @@ interface TrackerController : TrackerConfigurationInterface {
      * The tracker will start tracking again.
      */
     fun resume()
+
+    /**
+     * Adds user and session information to a URI.
+     *
+     * For example, calling decorateLink on `appSchema://path/to/page` will return:
+     *
+     *      `appSchema://path/to/page?_sp=userId.timestamp.sessionId.appId.platform.domainUserId`
+     *
+     * @param uri The URI to add the query string to
+     * @param parameters The parameters to include in the query string (defaults to all)
+     *
+     * @return Optional Uri:
+     * - null if [Session.userId] is null from `sessionContext(false)` being passed in [TrackerConfiguration]
+     * - otherwise, decorated Uri
+     */
+    fun decorateLink(
+        uri: Uri,
+        parameters: List<CrossDeviceParameter> = CrossDeviceParameter.values().toList()
```

Comment:
In the JS tracker, @igneel64 called this the `extendedCrossDomainLinker` and set the defaults to `sessionId` and `sourceId`. Would be great to do something similar here, maybe we could call the property `extendedParameters` (to make it clear that it these are not all the parameters being set in the URI) and by default only set the session and source ID?